tasks/bond_configuration.yml

```diff
@@ -13,12 +12,28 @@
   vars:
     bond_check: "{{ item | bond_check }}"
 
+- name: Create the nmconnection file for bond devices
+  become: true
+  template:
+    src: "bond_nmconnection.j2"
+    dest: "/etc/NetworkManager/system-connections/{{ item.device }}.nmconnection"
+    mode: "0600"
+  with_items: '{{ interfaces_bond_interfaces }}'
+  when:
+    - interfaces_use_nmconnection
```

Comment:
I'm a bit concerned about the performance implications of all of these if/else clauses. In large scale systems, each task can add considerable overhead.

Would it be possible to have a top-level include_tasks switch for nmcli vs non-nmcli? It's almost feeling like a different role at that point...

Reply:
> I'm a bit concerned about the performance implications of all of these if/else clauses. In large scale systems, each task can add considerable overhead.

Not sure, but wouldn't think it to have much of an impact. (atleast nothing near what inject_facts_as_vars had)

> Would it be possible to have a top-level include_tasks switch for nmcli vs non-nmcli? It's almost feeling like a different role at that point...

Yeah, not sure how to structure it. Small PoC for bond interfaces at eb4x/ansible-role-interfaces@59a385008c39825503d5cddfb9979fcf024f2593 but didn't like the way that went. But it led to the "smol" PoC which I do like.